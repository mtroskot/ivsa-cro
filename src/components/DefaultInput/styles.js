import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  default: {
    width: '100%',
    borderWidth: 1,
    borderColor: 'black',
    padding: 5,
    marginTop: 8,
    marginBottom: 8,
    borderRadius: 5
  },
  invalid: {
    backgroundColor: '#f9c0c0',
    borderColor: 'red'
  }
});

export default styles;
