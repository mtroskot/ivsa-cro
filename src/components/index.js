export { default as CustomButton } from 'src/components/CustomButton';
export { default as Error404 } from 'src/components/Error404';
export { default as FloatingLabelTextInput } from 'src/components/FloatingLabelTextInput';
export { default as IconButton } from 'src/components/IconButton';
export { default as PopupMessage } from 'src/components/PopupMessage';
export { default as KeyboardAvoidAndDismissView } from 'src/components/KeyboardAvoidAndDismissView';
export { default as ListItem } from 'src/components/ListItem';
export { default as Loader } from 'src/components/Loader';
export { default as PagingButtons } from 'src/components/PagingButtons';
