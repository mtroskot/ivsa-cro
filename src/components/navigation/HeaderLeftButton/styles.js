import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  backButton: {
    marginLeft: 15
  }
});

export default styles;
