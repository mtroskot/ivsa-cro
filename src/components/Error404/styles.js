import { StyleSheet } from 'react-native';
import { dimensions } from 'src/styles';
const { rem } = dimensions;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  errorText: {
    fontSize: 20 * rem,
    textAlign: 'center'
  },
  image: {
    height: 200 * rem,
    width: 200 * rem,
    backgroundColor: '#fff'
  }
});

export default styles;
