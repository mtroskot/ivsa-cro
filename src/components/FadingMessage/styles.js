import { StyleSheet } from 'react-native';
import { dimensions } from 'src/styles';
const { rem } = dimensions;

const styles = StyleSheet.create({
  popupMessageView: {
    flex: 1,
    zIndex: 2,
    backgroundColor: '#4c4c4c',
    padding: 11 * rem,
    borderRadius: 20 * rem,
    justifyContent: 'center',
    position: 'absolute',
    alignSelf: 'center'
  },
  message: {
    color: 'white',
    textAlign: 'center'
  },
  topPosition: {
    top: 120 * rem
  },
  bottomPosition: {
    bottom: 40 * rem
  }
});

export default styles;
