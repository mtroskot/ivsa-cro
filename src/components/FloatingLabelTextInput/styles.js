import { StyleSheet } from 'react-native';
import { dimensions } from 'src/styles';
const { rem } = dimensions;

const paddingLeft = 15 * rem;
const labelSize = 13 * rem;
const inputSize = 20 * rem;

const styles = StyleSheet.create({
  defaultInput: {
    fontSize: inputSize,
    borderColor: '#bbb',
    color: '#334144',
    flexGrow: 1,
    paddingLeft,
    marginTop: labelSize + labelSize * 0.1
  },
  container: {
    maxHeight: '40%',
    width: '90%',
    borderColor: '#DFE5EA',
    borderWidth: 1 * rem,
    alignSelf: 'center',
    borderRadius: 5 * rem,
    minHeight: 50 * rem,
    backgroundColor: '#F5F6F7',
    justifyContent: 'flex-end',
    paddingBottom: 5 * rem
  },
  floatingLabel: {
    position: 'absolute',
    left: paddingLeft,
    color: '#949EA0',
    fontSize: labelSize
  }
});

export default styles;
