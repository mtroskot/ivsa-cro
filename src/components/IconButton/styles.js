import { StyleSheet } from 'react-native';
import { dimensions } from 'src/styles';
const { rem } = dimensions;

const styles = StyleSheet.create({
  leftButton: {
    paddingRight: 30 * rem,
    marginLeft: 15 * rem
  },
  rightButton: {
    paddingLeft: 30 * rem,
    marginRight: 15 * rem
  },
  menu: {
    height: 6 * rem,
    width: 30 * rem
  },
  tabs: {
    height: 32 * rem,
    width: 32 * rem
  },
  backArrow: {
    height: 18 * rem,
    width: 10 * rem
  }
});
export default styles;
