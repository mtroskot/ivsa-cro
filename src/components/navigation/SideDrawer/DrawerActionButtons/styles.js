import { StyleSheet } from 'react-native';
import { dimensions } from 'src/styles';
const { rem } = dimensions;
const styles = StyleSheet.create({
  drawerItem: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10 * rem,
    backgroundColor: '#eee',
    marginTop: 10 * rem
  },
  drawerItemIcon: {
    marginRight: 10 * rem
  }
});

export default styles;
