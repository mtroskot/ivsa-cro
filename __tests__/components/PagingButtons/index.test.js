import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import { PagingButtons } from 'src/components';
import styles from 'src/components/PagingButtons/styles';

describe('PagingButtons wrapper', () => {
  it('PagingButtons renders with label and value', () => {
    const props = {
      pagingButtons: ['1', '2', '>'],
      currentPage: 0,
      onPress: jest.fn()
    };
    const wrapper = shallow(<PagingButtons {...props} />);
    expect(wrapper.first().type()).toEqual(View);
    expect(wrapper.find('View')).toHaveLength(2);
    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(3);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('text')
    ).toEqual('1');
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('tOpacityStyle')[1].backgroundColor
    ).toEqual(styles.buttonSelected.backgroundColor);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(1)
        .prop('text')
    ).toEqual('2');
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(1)
        .prop('tOpacityStyle')[1].backgroundColor
    ).toEqual(styles.buttonUnselected.backgroundColor);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(2)
        .prop('text')
    ).toEqual('>');
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(2)
        .prop('tOpacityStyle')[1].backgroundColor
    ).toEqual(styles.buttonUnselected.backgroundColor);
    expect(wrapper).toMatchSnapshot();
  });
});
