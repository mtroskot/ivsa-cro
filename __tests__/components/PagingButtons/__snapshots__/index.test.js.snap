// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagingButtons wrapper PagingButtons renders with label and value 1`] = `
<View
  style={
    Object {
      "flexGrow": 1,
      "justifyContent": "flex-end",
      "marginVertical": 39.473684210526315,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Memo(CustomButton)
      key="1"
      onPress={[Function]}
      tOpacityStyle={
        Array [
          Object {
            "backgroundColor": "#f4f4f4",
            "borderRadius": 9.868421052631579,
            "borderWidth": 1,
            "padding": 19.736842105263158,
          },
          Object {
            "backgroundColor": "#bababa",
          },
        ]
      }
      text="1"
    />
    <Memo(CustomButton)
      key="2"
      onPress={[Function]}
      tOpacityStyle={
        Array [
          Object {
            "backgroundColor": "#f4f4f4",
            "borderRadius": 9.868421052631579,
            "borderWidth": 1,
            "padding": 19.736842105263158,
          },
          Object {
            "backgroundColor": "#f4f4f4",
          },
        ]
      }
      text="2"
    />
    <Memo(CustomButton)
      key=">"
      onPress={[Function]}
      tOpacityStyle={
        Array [
          Object {
            "backgroundColor": "#f4f4f4",
            "borderRadius": 9.868421052631579,
            "borderWidth": 1,
            "padding": 19.736842105263158,
          },
          Object {
            "backgroundColor": "#f4f4f4",
          },
        ]
      }
      text=">"
    />
  </View>
</View>
`;
